src/drivers/gps/gps.cpp

```diff
@@ -526,15 +514,37 @@ GPS::print_info()
 	usleep(100000);
 }
 
+void
+GPS::publish()
+{
+	if (_gps_num == 1) {
+		if (_report_gps_pos_pub != nullptr) {
+			orb_publish(ORB_ID(vehicle_gps_1_position), _report_gps_pos_pub, &_report_gps_pos);
+
+		} else {
+			_report_gps_pos_pub = orb_advertise(ORB_ID(vehicle_gps_1_position), &_report_gps_pos);
```

Comment:
This should be using the orb_advertise_multi() or orb_advertise_auto() calls instead, see e.g. the MPU6000 driver.